.github/workflows/gh_pages.yml

```
@@ -1,30 +1,41 @@
 name: Github Pages
 on:
-  release:
-    types: [published]
+  push:
+    branches:
+      - master
 jobs:
   build-and-deploy:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v2.3.1
-      
+
       - name: Install Node (14.x)
         uses: actions/setup-node@v2
         with:
           node-version: '14.x'
 
       - name: Install Dependencies
         run: yarn install
-        
+
       - name: Build
         run: gulp
 
-      - name: Generate JSDoc
-        run: yarn jsdoc -c .jsdoc.json --verbose
+      - name: Checkout GH pages
+        uses: actions/checkout@v2
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git clone -b gh-pages "git@github.com:stellar/js-stellar-base.git" jsdoc
 
-      - name: Deploy to GH Pages
-        uses: JamesIves/github-pages-deploy-action@0f24da7de3e7e135102609a4c9633b025be8411b
-        with:
-          branch: gh-pages # The branch the action should deploy to.
-          folder: jsdoc # The folder the action should deploy.
+      - name: Generate JS docs
+        run: yarn docs
+
+      - name: Publish JS docs
+        uses: actions/checkout@v2
+        run: |
+          DESCRIPTION=`git describe --tags`
```

```suggestion
          DESCRIPTION=`git describe --abbrev=0 --tags`
```

If you don't use the `--abbrev=0` option, the DESCRIPTION will contain a little more information than just the latest tag.

```
git describe --abbrev=0 --tags  // returns "v6.0.1"
git describe --tags             // returns "v6.0.1-5-gd073684"
```

Is that your intention? The commit messages from branch [gh-pages](https://github.com/stellar/js-stellar-base/commits/gh-pages) are in the format `v6.0.1`.

---

Yeah, I should've mentioned that I did this intentionally. I'm obviously open to critique, but I really like that this creates a tighter relationship between the library and the docs. This was partially inspired by what you said above: tag commits can change.